pricing/src/main/scala/dev/atedeg/mdm/pricing/Actions.scala

```diff
@@ -0,0 +1,54 @@
+package dev.atedeg.mdm.pricing
+
+import java.time.LocalDateTime
+
+import cats.data.NonEmptyList
+import cats.syntax.all.*
+
+import dev.atedeg.mdm.pricing.utils.{ percent as _, * }
+import dev.atedeg.mdm.utils.*
+import dev.atedeg.mdm.utils.given
+
+@SuppressWarnings(Array("org.wartremover.warts.ListAppend"))
+def priceOrderLine(
+    priceList: PriceList,
+    clientPromotions: List[Promotion],
+    today: LocalDateTime,
+)(orderLine: IncomingOrderLine): PriceInEuroCents =
+  val (quantity, product) = (orderLine.quantity, orderLine.product)
+  val baseUnitPrice = priceList.priceList(product)
+  val activePromotions = clientPromotions.filter(_.expiryDate.isAfter(today))
+  val lines = activePromotions.flatMap(_.lines.toList).filter(_.product === product)
+  // Get the fixed discounts
+  val fixedDiscounts = lines.collect { case p @ PromotionLine.Fixed(_, _) => p }
```

Comment:
```suggestion
  val fixedDiscounts = lines.collect { case p: PromotionLine.Fixed => p }
```
the same applies before